### ISSUE TITLE
Fast pixel markers for CairoMakie 

### ISSUE THREAD

**Issue by asinghvi17** [JuliaPlots/Makie.jl#585]
This came up when I ran the orbit diagram example in CairoMakie - it's horrifically slow, even for a PNG surface, just because of the sheer number of drawing ops.

We should implement a "fast pixel" marker by manually creating and mutating an image surface, and then drawing that on to the screen's surface.

**Comment by SimonDanisch** [comment]
Yeah makes sense! We should move the FastPixel marker to AbstractPlotting anyways!
I just wrote @jkrumbiegel a lengthy explanation, how I really want to have all backend specific hacks still live in AbstractPlotting, so that AbstractPlotting is the only source of trueth and acts like all backends support the same features :D

**Comment by asinghvi17** [comment]
That makes sense...if we can get all backends to support a fast pixel marker type, that would be perfect.  I suppose the best way to do this would be to bin the scatter plot into rectangles, one per pixel?  Would that be achievable by using `project_position` on the input points, rounding the output to Int, and then filling the appropriate cells?

**Comment by SimonDanisch** [comment]
That sounds doable ;)

**Comment by ffreyer** [comment]
CairoMakie has been able to process FastPixel for a while
